src/scribe_data/checkquery.py

```diff
@@ -0,0 +1,442 @@
+
+
+def sparql_context(url: str) -> SPARQL.SPARQLWrapper:
+    """Configure a SPARQL context.
+
+    A context allows the execution of SPARQL queries.
+
+    Args:
+        url (str): a valid URL of a SPARQL endpoint.
+
+    Returns:
+        SPARQLWrapper: the context.
+    """
+    context = SPARQL.SPARQLWrapper(url)
+    context.setReturnFormat(SPARQL.JSON)
+    context.setMethod(SPARQL.POST)
+
+    return context
+
+
+def execute(
+    query: QueryFile, limit: int, context: SPARQL.SPARQLWrapper, tries: int = 3
+) -> dict:
+    """Execute a SPARQL query in a given context.
+
+    Args:
+        query (QueryFile): the SPARQL query to run.
+        limit (int): the maximum number of results a query should return.
+        context (SPARQLWrapper): the SPARQL context.
+        tries (int): the maximum number of times the query should be executed
+                    after failure.
+
+    Returns:
+        dict: results of the query.
+    """
+
+    def delay_in_seconds() -> int:
+        """How long to wait, in seconds, between executing repeat queries."""
+        return int(math.ceil(10.0 / math.sqrt(tries)))
+
+    if tries <= 0:
+        raise QueryExecutionException("Failed too many times.", query)
+
+    try:
+        context.setQuery(query.load(limit))
+        result = context.query().convert()
+        return result if result else []
+
+    except HTTPError:
+        time.sleep(delay_in_seconds())
+        return execute(query, limit, context, tries - 1)
+
+    except SPARQLExceptions.SPARQLWrapperException as err:
+        raise QueryExecutionException(err.msg, query) from err
+
+    except Exception as err:
+        raise QueryExecutionException(
+            f"{type(err).__name__} - {str(err)}", query
+        ) from err
+
+
+def check_sparql_file(query_file: str) -> Path:
+    """Check meta information of SPARQL query file.
+
+    Args:
+        query_file (str): the file to validate.
+
+    Returns:
+        Path: the validated file.
+    """
+    fpath = Path(query_file)
+
+    if not fpath.is_file():
+        raise argparse.ArgumentTypeError(f"Not a valid file path: {fpath}")
+
+    if fpath.suffix != ".sparql":
+        raise argparse.ArgumentTypeError(f"{fpath} does not have a '.sparql' extension")
+
+    return fpath
+
+
+def check_positive_number(value: str, err_msg: str) -> int:
+    """Ensure 'value' is a positive number.
+
+    Args:
+        value (str): the value to be validated.
+        err_msg (str): used when value fails validation.
+
+    Raises:
+        argparse.ArgumentTypeError
+
+    Returns:
+        int: the validated number.
+    """
+    try:
+        number = int(value)
+        if number >= 1:
+            return number
+    except ValueError:
+        pass
+
+    raise argparse.ArgumentTypeError(err_msg)
+
+
+def check_limit(limit: str) -> int:
+    """Validate the 'limit' argument.
+
+    Args:
+        limit (str): the LIMIT to be validated.
+
+    Raises:
```

Comment:
This is really nice. With the issue that we do for reworking the doc strings let's for sure also add in a `Raises` for files that need it! I added this to the sync notes.